components/SignInOrJoinFreeV2.js

```diff
@@ -103,7 +100,11 @@ class SignInOrJoinFreeV2 extends React.Component {
     if (window.location.search) {
       currentPath = currentPath + window.location.search;
     }
-    return encodeURIComponent(this.props.redirect || currentPath || '/');
+    let redirectUrl = this.props.redirect;
+    if (currentPath === '/create-accountv2' && redirectUrl === '/') {
+      redirectUrl = '/welcome-to-oc';
+    }
```

Comment:
`components/SignInOrJoinFreeV2.js` should ideally be agnostic about the rest of the website, or where it's called from. The component accepts a `redirect` prop, so it sounds to me like a way more simple solution is to pass `redirect="/welcome"` from the places where we want to redirect to the welcome page

Reply:
But here the `SignInOrJoinFreeV2.js` page is what dictates the actions based on the user choice (whether they chose to sign in or create a new account); the `SignInOrJoinFreeV2.js` is what determines whether the `SignInV2` or the `CreateProfileV2.js` is rendered depending on user action.

For the redirect prop here we pass the actual redirect; based on where the user was. For example they can be on the `/search` page and decides to "sign-in". So we need this `redirect` prop to store that information of where the user was. Overwriting this with `/welcome` isn't good as the user might decide to "sign-in" (if they already have an account). Only if they decide to join the platform (create a new profile) this `/welcome` route comes into play. Since the whole logic of switching between `/create-accountv2` and `/sign-inv2` is handled by `SignInOrJoinFreeV2.js` I am thinking it's best to place this logic here. WDYT? 🤔